main.tf

```diff
@@ -0,0 +1,53 @@
+module "file_system" {
+  source = "./modules/nas_file_system"
+
+  ###############################################################
+  #variables for file system
+  ##############################################################
+  protocol_type = "${var.protocol_type}"
+
+  create_file_system = "${var.create_file_system}"
+  storage_type       = "${var.storage_type}"
+  description        = "${var.file_system_description}"
+}
+
+module "access_group" {
+  source = "./modules/nas_access_group"
+
+  ###############################################################
+  #variables for access group
+  ##############################################################
+  access_group_name = "${var.access_group_name}"
+
+  create_access_group = "${var.create_access_group}"
+  type                = "${var.type}"
+  description         = "${var.access_group_description}"
+}
+
+module "access_rule" {
+  source = "./modules/nas_access_rule"
+
+  ###############################################################
+  #variables for access rule
+  ##############################################################
+  access_group_name = "${var.access_group_name}"
+
+  create_access_rule = "${var.create_access_rule}"
+  source_cidr_ip     = "${var.source_cidr_ip}"
+  rw_access_type     = "${var.rw_access_type}"
+  user_access_type   = "${var.user_access_type}"
+  priority           = "${var.priority}"
+}
+
+module "mount_target" {
+  source = "./modules/nas_mount_target"
+
+  ###############################################################
+  #variables for mount target
+  ##############################################################
+  create_mount_target = "${var.create_mount_target}"
+
+  access_group_name = "${var.access_group_name}"
+  file_system_id    = "${var.file_system_id}"
```

Comment:
I think line 51 should be `file_system_id    = "${var.create_file_system? module.file_system.this_file_system_id : var.file_system_id}"`